compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1298,11 +1298,16 @@ TR::Register *OMR::X86::TreeEvaluator::iternaryEvaluator(TR::Node *node, TR::Cod
    TR::Register *falseReg = cg->evaluate(falseVal);
    bool trueValIs64Bit = TR::TreeEvaluator::getNodeIs64Bit(trueVal, cg);
    TR::Register *trueReg  = TR::TreeEvaluator::intOrLongClobberEvaluate(trueVal, trueValIs64Bit, cg);
+   if (!node->isNotCollected()) 
+      trueReg->setContainsCollectedReference();
 
    // don't need to test if we're already using a compare eq or compare ne
    auto conditionOp = condition->getOpCode();
+   bool longCompareOn32bit = (conditionOp.isBooleanCompare() &&
+         (condition->getFirstChild()->getOpCode().isLong() || condition->getSecondChild()->getOpCode().isLong())
+         && (!TR::Compiler->target.is64Bit()));
```

Comment:
It might be more clear to use `TR::Compiler->target.is64Bit()`.
As the C++ compiler may not be able to fold this query to a constant, it may be worthy moving it as the first condition being checked, in which way the rest checks can be skipped on 64-bit.

Reply:
I *think* @0dvictor meant `TR::Compiler->target.is32Bit()` ???

Reply:
Sorry about the typo. I indeed meant `TR::Compiler->target.is32Bit()`.

Reply:
addressed.